warn/warn.go

```diff
@@ -413,6 +413,70 @@ func integerDivisionWarning(f *build.File, fix bool) []*Finding {
 	return findings
 }
 
+func unsortedDictItemsWarning(f *build.File, fix bool) []*Finding {
+	findings := []*Finding{}
+
+	compareItems := func(item1, item2 *build.KeyValueExpr) bool {
+		key1 := item1.Key.(*build.StringExpr)
+		key2 := item2.Key.(*build.StringExpr)
+		return key1.Value < key2.Value
+	}
+
+	build.Walk(f, func(expr build.Expr, stack []build.Expr) {
+		dict, ok := expr.(*build.DictExpr)
+		if !ok {
+			return
+		}
+		// do not process dictionaries nested within expressions that do not
+		// want dict items to be sorted
+		for i := len(stack) - 1; i >= 0; i-- {
```

Comment:
What if the dict itself has a comment? I think that can still be the case, e.g.

    d.update(
        # comment 1
        {a: b},  # comment 2
    )

  If I understand correctly, `stack` doesn't contain the current node.

Reply:
interesting, I could not find a way to associate a comment with a dictionary, which is why I had to use `stack` in the first place, but the examples you've provided actually manage to achieve this :D Added tests and code to make sure they are handled properly. Thanks for your help!